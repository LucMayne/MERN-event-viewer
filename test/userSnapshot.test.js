const { TextEncoder, TextDecoder } = require('util');
global.TextEncoder = TextEncoder;
global.TextDecoder = TextDecoder;

const request = require('supertest');
const express = require('express');
const app = express();
const router = require('../routes/users.js'); 

app.use(router);

// snapshot test
describe('Test the /login route', () => {
    test('respond with an error if incorrect info is entered', async () => {
        const response = await request(app)
            .post('/login')
            .send({ username: 'incorrect', password: 'incorrect' });
        // expect a 401 
        expect(response.statusCode).toBe(401);
        // create a snapshot of the response body
        expect(response.body).toMatchSnapshot();
    });
});
