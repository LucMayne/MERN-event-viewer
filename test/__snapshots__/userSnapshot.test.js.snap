// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the /login route respond with an error if incorrect info is entered 1`] = `
{
  "message": "Invalid credentials",
}
`;
